tests/integration/fixtures.py

```diff
@@ -332,15 +332,15 @@ def gpu_client(gpu_cluster):
 # client for all computations. otherwise, only connect to a client
 # when specified.
 @pytest.fixture(
-    scope="function" if SCHEDULER_ADDR is None else "session",
+    scope="function",
```

Comment:
When we use a session-wide client for external cluster testing, it ends up getting blown away with `dask.config.refresh()` while running `test_config.py`. Since there wasn't any immediate way to avoid this, I made this change so that we reconnect to the external cluster on each test, so that tests run after `test_config.py` are still able to use the external cluster as expected.

Reply:
I also wonder if `test_config` can be improved somehow to save the state of configs before the tests start and restore that state at the end of the test. 

Either way I didn't see a noticeable change in overall test runtimes with this change so it might be okay to improve later.